ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/beaconstate/BeaconStateInvariants.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.datastructures.state.beaconstate;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import java.util.List;
+import org.apache.tuweni.bytes.Bytes;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.ssz.backing.schema.SszPrimitiveSchemas;
+import tech.pegasys.teku.ssz.backing.schema.SszSchema;
+import tech.pegasys.teku.ssz.backing.view.SszPrimitives;
+import tech.pegasys.teku.ssz.sos.SszField;
+
+class BeaconStateInvariants {
+  // Schemas
+  static final SszSchema<SszPrimitives.SszUInt64> GENESIS_TIME_SCHEMA =
```

Comment:
BTW, may be it makes sense to move `SszPrimitives` internal classes to a separate package? I just always did static imports for them and didn't bother with such long names .

Reply:
I suppose it would be a little cleaner if we did move them out, but doesn't bother me too much either way.